tests/test_jsonparser.py

```diff
@@ -0,0 +1,19 @@
+import io
```

Comment:
Thank you for adding tests

Reply:
I had to make some small changes to pull out Page as a reusable class, maybe I didn't need to but I wanted some basic tests at least.